File: MBT-pptskills/assets/generate-placeholder.js
// 生成占位图的 Node.js 脚本
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

const width = 1024;
const height = 1024;

const canvas = createCanvas(width, height);
const ctx = canvas.getContext('2d');

// 背景色 #E0E0E0
ctx.fillStyle = '#E0E0E0';
ctx.fillRect(0, 0, width, height);

// 绘制中心文字 "Image Placeholder"
ctx.fillStyle = '#666666';
ctx.font = 'bold 60px Arial';
ctx.textAlign = 'center';
ctx.textBaseline = 'middle';
ctx.fillText('Image Placeholder', width / 2, height / 2);

// 保存为 PNG
const buffer = canvas.toBuffer('image/png');
const outputPath = path.join(__dirname, 'placeholder.png');
fs.writeFileSync(outputPath, buffer);

console.log('✅ Placeholder image created:', outputPath);

File: MBT-pptskills/assets/generate-placeholder.py
"""
生成占位图的 Python 脚本
使用 PIL/Pillow 创建 1024x1024 的灰底占位图
"""
from PIL import Image, ImageDraw, ImageFont
import os

# 创建 1024x1024 的图片
width, height = 1024, 1024
img = Image.new('RGB', (width, height), color='#E0E0E0')

# 绘制文字
draw = ImageDraw.Draw(img)
text = "Image Placeholder"

# 尝试使用系统字体，如果失败则使用默认字体
try:
    # Windows 常见字体路径
    font = ImageFont.truetype("C:/Windows/Fonts/arial.ttf", 60)
except:
    # 使用默认字体
    font = ImageFont.load_default()

# 计算文字位置（居中）
bbox = draw.textbbox((0, 0), text, font=font)
text_width = bbox[2] - bbox[0]
text_height = bbox[3] - bbox[1]
x = (width - text_width) / 2
y = (height - text_height) / 2

# 绘制文字
draw.text((x, y), text, fill='#666666', font=font)

# 保存图片
output_path = os.path.join(os.path.dirname(__file__), 'placeholder.png')
img.save(output_path, 'PNG')

print(f'Placeholder image created: {output_path}')
